pkg/mimir/mimir.go

```diff
@@ -490,7 +491,11 @@ type Mimir struct {
 }
 
 // New makes a new Mimir.
-func New(cfg Config) (*Mimir, error) {
+func New(cfg Config, reg prometheus.Registerer) (*Mimir, error) {
+	if reg == nil {
+		return nil, errors.New("no Prometheus registerer provided")
+	}
```

Comment:
It's common that `nil` registrerer is a valid value. Why not here? (Not that we would use it like that)

Reply:
On the other hand, if you forget to set it then integration tests will fail. Rethinking it, I will accept `nil`.

Reply:
If someone deliberately passes in a `nil` value, then they can also deliberately pass an empty registry, so I wouldn't care about that.

Reply:
Very good point, Oleg!

Reply:
Addressed in Addressed in https://github.com/grafana/mimir/pull/2750/commits/ae2a9b3eb1b476ea03bf96c6f653bd07121b9b33